crates/red_knot/tests/cli.rs

```diff
@@ -51,6 +51,141 @@ print(sys.last_exc)
     Ok(())
 }
 
+/// Paths specified on the CLI are relative to the current working directory and not the project root.
+///
+/// We test this by adding an extra search path from the CLI to the libs directory when
+/// running the CLI from the child directory (using relative paths).
+///
+/// Project layout:
+/// ```
+///  - libs
+///    |- utils.py
+///  - child
+///    | - test.py
+/// - pyproject.toml
+/// ```
+///
+/// And the command is run in the `child` directory.
+#[test]
+fn cli_arguments_are_relative_to_the_current_directory() -> anyhow::Result<()> {
+    let tempdir = TempDir::new()?;
+
+    let project_dir = tempdir.path();
+
+    let libs = project_dir.join("libs");
+    std::fs::create_dir_all(&libs).context("Failed to create `libs` directory")?;
+
+    let child = project_dir.join("child");
+    std::fs::create_dir(&child).context("Failed to create `child` directory")?;
+
+    std::fs::write(
+        tempdir.path().join("pyproject.toml"),
+        r#"
+[tool.knot.environment]
+python-version = "3.11"
+"#,
+    )
+    .context("Failed to write `pyproject.toml`")?;
+
+    std::fs::write(
+        libs.join("utils.py"),
+        r#"
+def add(a: int, b: int) -> int:
+    a + b
+"#,
+    )
+    .context("Failed to write `utils.py`")?;
+
+    std::fs::write(
+        child.join("test.py"),
+        r#"
+from utils import add
+
+stat = add(10, 15)
+"#,
+    )
+    .context("Failed to write `child/test.py`")?;
+
```

Comment:
Since this is a slightly involved test setup, I think I would appreciate something like
```suggestion

    // Make sure that the CLI fails when the `libs` directory is not in the search path.
    insta::with_settings!({filters => vec![(&*tempdir_filter(&tempdir), "<temp_dir>/")]}, {
        assert_cmd_snapshot!(knot().current_dir(&child), @r#"
        success: false
        exit_code: 1
        ----- stdout -----
        error[lint:unresolved-import] <temp_dir>/child/test.py:2:1 Cannot resolve import `utils`

        ----- stderr -----
        "#);
    });

```